lib/logstash/inputs/heartbeat.rb

```diff
@@ -43,23 +47,27 @@ def run(queue)
     sequence = 0
 
     Stud.interval(@interval) do
-      if @message == "epoch"
-        event = LogStash::Event.new("clock" => Time.now.to_i, "host" => @host)
-      elsif @message == "sequence"
-        event = LogStash::Event.new("clock" => sequence, "host" => @host)
-        sequence += 1
-      else
-        event = LogStash::Event.new("message" => @message, "host" => @host)
-      end
-      
+      sequence += 1
+      event = generate_message(sequence)
       decorate(event)
       queue << event
-
+      break if sequence == @count
     end # loop
 
   end # def run
 
   public
+  def generate_message(sequence)
+    if @message == "epoch"
+      LogStash::Event.new("clock" => Time.now.to_i, "host" => @host)
+    elsif @message == "sequence"
+      LogStash::Event.new("clock" => sequence, "host" => @host)
+    else
+      LogStash::Event.new("message" => @message, "host" => @host)
+    end
+  end
```

Comment:
Maybe its a little yak shaving but I would move the conditionals into a predicate method and/or defining constants for the type of messages.

```ruby
epoch?
sequence?
```